guide/popular-topics/builders.md

```diff
@@ -0,0 +1,132 @@
+
+```js {2,4-5}
+const { hyperlink, hideLinkEmbed } = require('@discordjs/builders');
+const url = 'https://discord.js.org/';
+
+const link = hyperlink(url);
+const hiddenEmbed = hideLinkEmbed(url);
+```
+
+### Codeblocks
+
+You can use `inlineCode()` and `codeBlock()` to turn a string into an inline codeblock or a regular codeblock with or without syntax highlighting.
+
+```js {2,4-6}
+const { inlineCode, codeBlock } = require('@discordjs/builders');
+const jsString = 'const value = true;';
+
+const inline = inlineCode(jsString);
+const codeblock = codeBlock(jsString);
+const highlighted = codeBlock('js', jsString);
+```
+
+### Timestamps
+
+With `time()` you can format UNIX timestamps and dates into a Discord timestring.
+
+```js {2,4-5}
+const { time } = require('@discordjs/builders');
+const date = new Date();
+
+const timeString = time(date);
+const relative = time(date, 'R');
+```
+
+### Mentions
+
+The Formatters also contain various methods to format Snowflakes into mentions.
+
+```js {2,4-7}
+const { userMention, membermention, channelMention, roleMention } = require('@discordjs/builders');
+const id = '123456789012345678';
+
+const user = userMention(id);
+const nickname = memberMention(id);
+const channel = channelMention(id);
+const role = roleMention(id);
+```
+
+## Slash command builders
+
+The slash command builder is a utility class to quickly build slash commands without having to manually construct objects.
+
+### Commands
+
+First we'll build a simple slash command using the builder and save the raw data to a variable, which can then later be used to send the data to the Discord API.
+
+```js
+const { SlashCommandBuilder } = require('@discordjs/builders');
+
+// Create a slash command builder
+const command = new SlashCommandBuilder().setName('ping').setDescription('Replies with Pong!');
+
+// Get the raw data that can be sent to Discord
+const rawData = command.toJSON();
+```
+
+### Options
+
+This is a command with a user option.
+
+```js {4-6}
+const { SlashCommandBuilder } = require('@discordjs/builders');
+
+const command = new SlashCommandBuilder()
+	.setName('info')
+	.setDescription('Get info about a user!')
+	.addUserOption(option => option.setName('user').setDescription('The user'));
+
+const rawData = command.toJSON();
+```
+
+### Subcommands
+
+And this is a command containing two subcommands.
+
+```js {6-14}
+const { SlashCommandBuilder } = require('@discordjs/builders');
+
+const command = new SlashCommandBuilder()
+	.setName('info')
+	.setDescription('Get info about a user or a server!')
+	.addSubcommand(subcommand =>
+		subcommand
+			.setName('user')
+			.setDescription('Info about a user')
+			.addUserOption(option => option.setName('target').setDescription('The user')))
+	.addSubcommand(subcommand =>
+		subcommand
+			.setName('server')
+			.setDescription('Info about the server'));
+
+const rawData = command.toJSON();
```

Comment:
```suggestion
```js {4-12}
const command = new SlashCommandBuilder()
	.setName('info')
	.setDescription('Get info about a user or a server!')
	.addSubcommand(subcommand =>
		subcommand
			.setName('user')
			.setDescription('Info about a user')
			.addUserOption(option => option.setName('target').setDescription('The user')))
	.addSubcommand(subcommand =>
		subcommand
			.setName('server')
			.setDescription('Info about the server'));
```